apps/signal/51-function-call-effect/src/app/user.service.ts

```diff
@@ -1,10 +1,10 @@
-import { Injectable, signal } from '@angular/core';
+import { Injectable, signal, untracked } from '@angular/core';
 
 @Injectable({ providedIn: 'root' })
 export class UserService {
   name = signal('Thomas');
 
   log(message: string) {
-    console.log(`${this.name()}: ${message}`);
+    console.log(`${untracked(() => this.name())}: ${message}`);
```

Comment:
I'm not sure, i'll put the untrack here. It might make sense to track name for some other use case. 

I would use the untrack inside the effect since it's where we don't want to track what's inside the function. And for instance, the log can be a library you don't own

Reply:
My initial commit was the minimal change to fix the bug.

Based on this feedback, I pushed another commit to move the fix to the component.